packages/core/src/pubsub/pubsub.ts

```diff
@@ -98,4 +110,21 @@ export class Pubsub extends Observable<PubsubMessage> {
         },
       })
   }
+
+  /**
+   * Called periodically and ensures that if we haven't published a pubsub message in too long,
+   * we'll publish one so that we never go longer than MAX_PUBSUB_PUBLISH_INTERVAL without
+   * publishing a pubsub message.  This is to work around a bug in IPFS where peer connections
+   * get dropped if they haven't had traffic in too long.
+   */
+  publishPubsubKeepaliveIfNeeded(): void {
+    const now = Date.now()
+    if (now - this.lastPublishedMessageDate < this.maxPubsubPublishInterval / 2) {
+      // We've published a message recently enough, no need to publish another
+      return
+    }
```

Comment:
Why do it like this when we can just trigger it on a regular interval from `setInterval`?

Reply:
You mean just unilaterally publish a message on an interval?  That would create additional pubsub traffic unnecessarily for a well-used node.  I'm trying to only publish these keepalive messages if the node isn't publishing other messages from normal use.  So basically only idle nodes will generate keepalive messages

Reply:
Oh I see! Makes sense ty!